kafka/consumer_test.go

```diff
@@ -295,6 +295,9 @@ func TestConsumerAssignment(t *testing.T) {
 		if err.(Error).Code() != ErrTimedOut {
 			t.Errorf("Expected ReadMessage to fail with ErrTimedOut, not %v", err)
 		}
+		if !err.(Error).IsTimeout() {
+			t.Errorf("Expected ReadMessage to fail with a timeout error, not %v", err)
```

Comment:
Maybe add a case for the success case too. E.g., from the TestConsumerAPIs dry-test (doesn't require a broker), you could call ReadMessage() with a low timeout and verify it is a timeout.

Reply:
I'm a bit unsure if I got this right since the case in `TestConsumerAssignment` is also a success case (ie it's a timeout). 
I added a test in TestConsumerAPIs, hopefully that captures the intent of this comment.